### PR TITLE
Add files

### PR DESCRIPTION
**Pull request by martinyde**
https://leantime.itkdev.dk/#/tickets/showTicket/2117

- Cleanup default module code
- Adds /metrics endpoint
- Implements incBy counter
- Reformat default code
- Remove unused code

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+See [keep a changelog] for information about writing changes to this log.
+
+## [1.0.0]
+
+* Adds /metrics endpoint
+* Implements incBy counter
```
Comment (cableman) [inline CHANGELOG.md]: Please use keep-a-change-log format - and the inline links for versions. See their log: https://github.com/olivierlacan/keep-a-changelog/blob/main/CHANGELOG.md?plain=1

File: Http/Controllers/ItkPrometheusController.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+class ItkPrometheusController extends Controller
+{
+    private PrometheusService $service;
+
+    public function __construct(PrometheusService $service)
+    {
+      $this->service = $service;
+    }
+
+  /**
```
Comment (cableman) [inline Http/Controllers/ItkPrometheusController.php]: Indention wrong...

File: Http/Controllers/ItkPrometheusController.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    public function create()
+    {
+        return view('itkprometheus::create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     * @param  Request $request
+     * @return Response
```
Comment (cableman) [inline Http/Controllers/ItkPrometheusController.php]: Normally we have an empty line between the different doc block types. But I do not known what the coding standard for freescout is.

File: Http/Controllers/ItkPrometheusController.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    public function store(Request $request)
+    {
+    }
```
Comment (cableman) [inline Http/Controllers/ItkPrometheusController.php]: Empty function? If required here, please make note that it is left empty.

File: Http/Controllers/ItkPrometheusController.php [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+    /**
+     * Remove the specified resource from storage.
+     * @return Response
+     */
+    public function destroy()
+    {
+    }
```
Comment (cableman) [inline Http/Controllers/ItkPrometheusController.php]: Same here as with `store`

File: Providers/ItkPrometheusServiceProvider.php [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        $this->registerConfig();
+        $this->registerViews();
+        $this->registerFactories();
+        $this->loadMigrationsFrom(__DIR__ . '/../Database/Migrations');
+        $this->hooks();
```
Comment (cableman) [inline Providers/ItkPrometheusServiceProvider.php]: Why have this call to an empty `hooks()` function... if real need please comment why

File: Public/js/laroute.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+(function () {
```
Comment (cableman) [inline Public/js/laroute.js]: Why is this JS code need and what do it do?

File: Service/PrometheusService.php [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  /**
+   * @param \Prometheus\CollectorRegistry $registry
+   */
+  public function __construct (CollectorRegistry $registry)
+  {
+    $this->collectorRegistry = $registry->getDefault();
+  }
+
+  /**
+   * @return string
```
Comment (cableman) [inline Service/PrometheusService.php:26]: Function comment would be nice to have. I think it renders the metrics end-point for scraping ?

**Review by cableman — CHANGES_REQUESTED** [review]
Please add return types to all functions

File: Http/Controllers/ItkPrometheusController.php (hunk 2 above)
Comment (martinyde) [reply Http/Controllers/ItkPrometheusController.php]: Fixed

File: CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [unreleased]
+
+## [1.0.0] - 2024-09-23
```
Comment (cableman) [inline CHANGELOG.md]: You need to make the links in the bottom to male the [version] into links. See https://github.com/itk-dev/economics/blob/develop/CHANGELOG.md?plain=1#L340

File: Service/PrometheusService.php [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+   * @param array $labels
+   *   The related labels.
+   * @param int $count
+   *   the count to increase by.
+   *
+   * @return void
+   * @throws \Prometheus\Exception\MetricsRegistrationException
+   *
+   */
+  public function incCounterBy(array $counter, array $labels, int $count = 1): void {
```
Comment (cableman) [inline Service/PrometheusService.php:54]: All functions need the return types specified link this one has `: void`

**Review by cableman — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 9 above)
Comment (martinyde) [reply CHANGELOG.md]: Removed version

**Review by cableman — APPROVED** [review]

**Pull request merged**